File: MinimumNumberofArrowstoBurstBalloons.py
class Solution:
    def findMinArrowShots(self, points) -> int:
        if not points:
            return 0
        points.sort()
        print(points)
        count = 1
        end = points[0][1]
        for i in range(1, len(points)):
            if points[i][0] <= end:
                end = min(end, points[i][1])
            else:
                count += 1
                end = points[i][1]
        return count


if __name__ == '__main__':
    nums = [int(num) for num in input().split()]
    points = [[nums[i], nums[i + 1]] for i in range(0, len(nums), 2)]
    solution = Solution()
    print(solution.findMinArrowShots(points))

File: MinimumNumberofArrowstoBurstBalloons.cpp
#include<iostream>
#include<vector>
#include<stack>
#include<algorithm>

using namespace std;

class Solution {
public:
    int findMinArrowShots(vector<pair<int, int> >& points) {
        int n = points.size();
        int count = 0;
        sort(points.begin(), points.end());
        while(!points.empty()) {
        	int left = INT_MIN, right = INT_MAX;
        	vector<pair<int, int> > temp;
        	for(auto point : points) {
        		if(point.first > right || point.second < left) temp.push_back(point);
        		else {
        			cout<<point.first<<","<<point.second<<"  ";
        			left = max(point.first, left);
        			right = min(point.second, right);
				}
			}
			cout<<endl;
			points = temp;
			count++;
		}
		return count;
    }
};

int main() {
	
	int n;
	cin>>n;
	
	vector<pair<int, int> > points;
	for(int i = 0; i < n; i++) {
		int a, b;
		cin>>a>>b;
		points.push_back(make_pair(a, b));
	}
	
	Solution *solution = new Solution();
	cout<<solution->findMinArrowShots(points);
	
	return 0;
	
}
